src/librustc_mir/borrow_check/diagnostics/region_errors.rs

```diff
@@ -728,13 +701,15 @@ impl<'tcx> RegionInferenceContext<'tcx> {
             ..
         } = errci;
 
-        let mut diag = infcx.tcx.sess.struct_span_err(*span, "lifetime may not live long enough");
+        let mut diag =
+            mbcx.infcx.tcx.sess.struct_span_err(*span, "lifetime may not live long enough");
 
-        let mir_def_name = if infcx.tcx.is_closure(*mir_def_id) { "closure" } else { "function" };
+        let mir_def_name =
+            if mbcx.infcx.tcx.is_closure(mbcx.mir_def_id) { "closure" } else { "function" };
```

Comment:
`MirBorrowckCtxt` should probably get a method that gets a description for the current item that's used everywhere. We have this sort of logic repeated in various places and often forget to handle generators and async functions. (cc #67687)

Reply:
Should we just as a query for that? It seems like there are these sorts of fns everywhere.

Reply:
I don't think it makes much sense as a query, since it's only for error reporting. It might make more sense being available more widely than just borrow checking.

Reply:
Right, that's what I meant.

Do you want me to do that in this PR?

Reply:
I think I would prefer it in another PR (maybe the next PR?).

Reply:
Sure, it might need to be the one after that, as the next PR is already ready (#67476) and I need to look for all the places where such a function would be used... would `librustc` be an appropriate place for something like this to live?

Reply:
I opened https://github.com/rust-lang/rust/pull/67742